Library/AppEnvironment.swift

```diff
@@ -124,15 +124,14 @@ public struct AppEnvironment {
     debounceInterval: DispatchTimeInterval = AppEnvironment.current.debounceInterval,
     device: UIDeviceType = AppEnvironment.current.device,
     facebookAppDelegate: FacebookAppDelegateProtocol = AppEnvironment.current.facebookAppDelegate,
-    isRegisteredForPushNotifications: SignalProducer<Bool, NoError> =
-      AppEnvironment.current.isRegisteredForPushNotifications,
     isVoiceOverRunning: @escaping (() -> Bool) = AppEnvironment.current.isVoiceOverRunning,
     koala: Koala = AppEnvironment.current.koala,
     language: Language = AppEnvironment.current.language,
     launchedCountries: LaunchedCountries = AppEnvironment.current.launchedCountries,
     liveStreamService: LiveStreamServiceProtocol = AppEnvironment.current.liveStreamService,
     locale: Locale = AppEnvironment.current.locale,
     mainBundle: NSBundleType = AppEnvironment.current.mainBundle,
+    pushRegistrationType: PushRegistrationType.Type = AppEnvironment.current.pushRegistrationType,
```

Comment:
Does it make more sense to declare this as `pushRegistration: PushRegistrationType.Type`? The object is a `pushRegistration` object of type `PushRegistrationType`. Then we can call it like `AppEnvironment.current.pushRegistration.register()`

Reply:
Well I chose to use the type because there doesn't really need to be any instance/state retained for this, they're just functions that execute and end. The type is just to group the functions and make it easier to mock behind the protocol.

Reply:
Fair enough 👍